lib/vector-core/src/config/mod.rs

```diff
@@ -62,6 +62,12 @@ impl Input {
 pub struct Output {
     pub port: Option<String>,
     pub ty: DataType,
+
+    /// NOTE: schema definitions are currently ignored for non-log events. In the future, we can
+    /// change `DataType` to keep track of schema data internally (e.g. keep one for the `Log` or
+    /// `Metric` variants, and two for the `Any` variant, one for log events and one for metrics).
+    /// Alternatively, we could update this field to `log_schema_definition` and add a new
+    /// `metric_schema_definition` as well.
```

Comment:
consider renaming the field to `log_schema_definition`, just to make this more clear